packages/cypress/resources/features/refapp-3.x/06-patient-search/patient-search.feature

```diff
@@ -9,5 +9,5 @@ Feature: Patient Search
     Then the result should be "<result>"
     Examples:
       | patientName  | result            |
-      | John Doe     | John Doe          |
+      | Richard Jones   | Richard Jones    |
```

Comment:
There's an issue with the patient registration API which breaks the patient search.
To reproduce: 
1. Create a patient using the API
2. Try to search for the patient
3. The search will break because it is sending `null` patients in the response. 

I will report this issue separately. Added this as a temporary fix until we resolve the issue. I assumed that demo patients which already come with the container don't change frequently. 
I'm not super clear on how the demo data is being handled. Are there any chances of getting them to change? How frequently do we update them and how do we update them?

Reply:
> There's an issue with the patient registration API which breaks the patient search. To reproduce:
> 
> 1. Create a patient using the API
> 2. Try to search for the patient
> 3. The search will break because it is sending `null` patients in the response.
> 
> I will report this issue separately. Added this as a temporary fix until we resolve the issue. I assumed that demo patients which already come with the container don't change frequently.

Nicely explained @jayasanka-sack 


> I'm not super clear on how the demo data is being handled. Are there any chances of getting them to change?

Demo data is added either as `.cvs` or `.json` or `.zip` files to the openmrs-distro-reference application.  
REF: Locations at in RefApp 3.x -> https://github.com/openmrs/openmrs-distro-referenceapplication/blob/main/distro/configuration/locations/locations.csv



> How frequently do we update them and how do we update them?

There ain't a straight timeframe of when a change might be done.  It depends on the issue a developer is solving.  Am just curious that when developers finally pick interest in why QA is breaking, it will be disappointing finding out that demo data is the cause of the failure.

Reply:
Seems the issue isn't reproducible anymore. I updated the test. 
Btw, we depend on demo data a lot. Ex: Location IDs, Identifier types, visit types, Encounters, concepts, etc. To avoid breaking tests we should integrate the tests to be run when we open a PR to the `openmrs-distro-referenceapplication` (including other repos) once we figure out a way to run tests against PRs.

Reply:
> To avoid breaking tests we should integrate the tests to be run when we open a PR to the `openmrs-distro-referenceapplication` (including other repos) once we figure out a way to run tests against PRs.

This will be very helpful.  Part of it was done with https://ci.openmrs.org/browse/REFAPP-D3E